packages/@aws-cdk/aws-ecs-patterns/lib/ecs/load-balanced-ecs-service.ts

```diff
@@ -43,32 +58,44 @@ export class LoadBalancedEc2Service extends LoadBalancedServiceBase {
    */
   public readonly service: ecs.Ec2Service;
 
+  /**
+   * Constructs a new instance of the LoadBalancedEc2Service class.
+   */
   constructor(scope: cdk.Construct, id: string, props: LoadBalancedEc2ServiceProps) {
     super(scope, id, props);
 
-    const taskDefinition = new ecs.Ec2TaskDefinition(this, 'TaskDef', {});
+    const taskDefinition = new ecs.Ec2TaskDefinition(this, 'TaskDef', {
+      executionRole: props.executionRole,
+      taskRole: props.taskRole
+    });
 
-    const container = taskDefinition.addContainer('web', {
+    const containerName = props.containerName !== undefined ? props.containerName : 'web';
+    const container = taskDefinition.addContainer(containerName, {
       image: props.image,
+      cpu: props.cpu,
       memoryLimitMiB: props.memoryLimitMiB,
       memoryReservationMiB: props.memoryReservationMiB,
       environment: props.environment,
-      logging: this.logDriver,
+      logging: this.logDriver
     });
-
     container.addPortMappings({
       containerPort: props.containerPort || 80
     });
 
-    const assignPublicIp = props.publicTasks !== undefined ? props.publicTasks : false;
-    const service = new ecs.Ec2Service(this, "Service", {
+    this.service = new ecs.Ec2Service(this, "Service", {
       cluster: props.cluster,
-      desiredCount: props.desiredCount || 1,
+      desiredCount: this.desiredCount,
       taskDefinition,
-      assignPublicIp
+      assignPublicIp: this.assignPublicIp,
+      serviceName: props.serviceName
     });
+    this.addServiceAsTarget(this.service);
+  }
 
-    this.service = service;
-    this.addServiceAsTarget(service);
+  /**
+   * Add an additional non-essential container to the service.
+   */
+  public addAdditionalContainer(id: string, props: ecs.ContainerDefinitionOptions) {
```

Comment:
Was the `addAdditionalContainer` something we agreed upon as a public API? Just wondering if we're not using simply `addContainer` bc it's an API on task def -- i feel like it's a simpler name.

Reply:
The intent of addAdditionalContainer came from a discussion I had with @nathanpeck. I wanted to distinguish that this method should be used to add additional containers, not to add the essential container to a task def.